### PR TITLE
Feature/#307 add open suse to travis and docs

### PR DESCRIPTION
**Pull request by name213**

### CODE REVIEW

**Comment by name213** [comment]
@gavv seems that I can't add you as a reviewer so I'm tagging you here

**Comment by gavv** [comment]
https://github.com/roc-project/roc/issues/307

**Comment by gavv** [comment]
Thanks!

A few notes:

* User cookbook is missing installation of `libuv-devel` and `intltool`.

* A typo: `cputtest` -> `cpputest`. Please test build instructions carefully!

* Maybe remove `-n` from zypper in user cookbook? Installing without confirmation may be inconvenient I think.

* Let's change "openSUSE Linux" to something more specific, like: "openSUSE Leap and later".

* Nitpick: let's move openSUSE from the end of the list right after CentOS, to group rpm-based distros together.

* Could you please also update these two pages too? https://roc-project.github.io/roc/docs/portability/supported_platforms.html and https://roc-project.github.io/roc/docs/development/continuous_integration.html

**Comment by gavv** [comment]
Here is the new image on docker hub: https://hub.docker.com/r/rocproject/travis-opensuse

We should wait until docker hub will build it, it usually takes a few hours because of the queues.

**Comment by name213** [comment]
Great, pushed requested changes

File: docs/sphinx/building/user_cookbook.rst [hunk 1]
```diff
@@ -121,6 +121,34 @@ Centos 7 and later
     # install libraries, tools, and PulseAudio modules
     $ sudo scons -Q --enable-pulseaudio-modules --build-3rdparty=libuv,openfec,pulseaudio,cpputest install
 
+openSUSE Leap and later
+------------
```
Comment (gavv) [inline docs/sphinx/building/user_cookbook.rst]: The line (`------------`) should be same length as the header:

```
openSUSE Leap and later
-----------------------
```

(this is the requirement of sphinx)

File: docs/sphinx/development/continuous_integration.rst [hunk 2]
```diff
@@ -37,6 +37,7 @@ rocproject/travis-ubuntu:14.04      ubuntu:14.04          x86_64        gcc-4.4,
 rocproject/travis-debian            debian:stable         x86_64        distro default
 rocproject/travis-fedora            fedora:latest         x86_64        distro default
 rocproject/travis-centos            centos:latest         x86_64        distro default
+rocproject/travis-opensuse          leap:latest           x86_64        distro default
```
Comment (gavv) [inline docs/sphinx/development/continuous_integration.rst]: `leap:latest` -> `opensuse/leap:latest`

File: docs/sphinx/portability/supported_platforms.rst [hunk 3]
```diff
@@ -32,6 +32,7 @@ Linux distributions:
 * Debian (stable)
 * Fedora (latest)
 * CentOS (latest)
+* openSUSE Leap (latest)
```
Comment (gavv) [inline docs/sphinx/portability/supported_platforms.rst]: `openSUSE Leap (latest)` -> `openSUSE (Leap)`

File: .travis.yml [hunk 4]
```diff
@@ -29,6 +29,7 @@ jobs:
         - docker pull rocproject/travis-centos
         - docker pull rocproject/travis-archlinux
         - docker pull rocproject/travis-alpine
+        - docker pull rocproject/travis-opensuse
```
Comment (gavv) [inline .travis.yml]: Nitpick: please place opensuse right after centos in this file as well (here and below).

File: .travis.yml [hunk 5]
```diff
@@ -38,6 +39,7 @@ jobs:
         - scripts/travis/docker.sh rocproject/travis-centos scripts/travis/linux-x86_64/centos.sh
         - scripts/travis/docker.sh rocproject/travis-archlinux scripts/travis/linux-x86_64/archlinux.sh
         - scripts/travis/docker.sh rocproject/travis-alpine scripts/travis/linux-x86_64/alpine.sh
+        - scripts/travis/docker.sh rocproject/travis-alpine scripts/travis/linux-x86_64/opensuse.sh
```
Comment (gavv) [inline .travis.yml]: `travis-alpine` -> `travis-opensuse`

**Comment by gavv** [comment]
It seems that the `develop` build is now broken after rebuilding docker images. I'll look into it tomorrow.

After fixing all the issues, please also squash all commits into one and rebase on fresh develop.

**Comment by gavv** [comment]
And it seems that you forgot to add executable permission to `opensuse.sh`.

**Comment by name213** [comment]
Fixed exec permission but looks like alpine still fails

**Comment by gavv** [comment]
I've fixed the alpine issue, so the build is passing now. Please ping me when you'll fix the remaining issues and squash commits.

**Comment by name213** [comment]
Hey, squashed and  fixed remaining issues

**Comment by gavv** [comment]
Thanks!

**Pull request merged**